pydantic_core/core_schema.py

```diff
@@ -1900,17 +1900,17 @@ def new_class_schema(
     ref: str | None = None,
     extra: Any = None,
     serialization: SerSchema | None = None,
-) -> NewClassSchema:
+) -> ModelSchema:
     """
-    Returns a schema that matches a new class, e.g.:
+    Returns a schema that matches a model, e.g.:
```

Comment:
Worth adding a bit more description here, something like:

```
a model schema generally contains a typed-dict schema, it will run the typed dict validator, then
create a new class and set the dict and fields set returned from the typed dict validator to `__dict__` and `__fields_set__` respectively.
```

Does that make sense?